src/start_menu.c

```diff
@@ -0,0 +1,1556 @@
+    startMenu->gfxResObj[GF_GFX_RES_TYPE_ANIM] = AddCellOrAnimResObjFromNarc(startMenu->gfxResMan[GF_GFX_RES_TYPE_ANIM], NARC_a_0_1_4, 63, TRUE, 0, GF_GFX_RES_TYPE_ANIM, HEAP_ID_FIELD);
+    sub_0200ADA4(startMenu->gfxResObj[GF_GFX_RES_TYPE_CHAR]);
+    sub_0200B00C(startMenu->gfxResObj[GF_GFX_RES_TYPE_PLTT]);
+    sub_0200A740(startMenu->gfxResObj[GF_GFX_RES_TYPE_PLTT]);
+    CreateSpriteResourcesHeader(&startMenu->spriteResourcesHeader, 0, 0, 0, 0, -1, -1, 0, 0, startMenu->gfxResMan[GF_GFX_RES_TYPE_CHAR], startMenu->gfxResMan[GF_GFX_RES_TYPE_PLTT], startMenu->gfxResMan[GF_GFX_RES_TYPE_CELL], startMenu->gfxResMan[GF_GFX_RES_TYPE_ANIM], NULL, NULL);
+
+    SpriteTemplate spriteTemplate;
+    spriteTemplate.spriteList = startMenu->spriteList;
+    spriteTemplate.header = &startMenu->spriteResourcesHeader;
+    spriteTemplate.position.z = 0;
+    spriteTemplate.scale.x = FX32_ONE;
+    spriteTemplate.scale.y = FX32_ONE;
+    spriteTemplate.scale.z = FX32_ONE;
+    spriteTemplate.rotation = 0;
+    spriteTemplate.priority = 0;
+    spriteTemplate.whichScreen = NNS_G2D_VRAM_TYPE_2DMAIN;
+    spriteTemplate.heapId = HEAP_ID_FIELD;
+    spriteTemplate.header = &startMenu->spriteResourcesHeader;  // second assign is necessary to match
+    spriteTemplate.position.x = FX32_CONST(100);
+    spriteTemplate.position.y = FX32_CONST(144);
+    startMenu->cursorSprite = CreateSprite(&spriteTemplate);
+    Set2dSpriteAnimActiveFlag(startMenu->cursorSprite, TRUE);
+    GfGfx_EngineATogglePlanes(GX_PLANEMASK_OBJ, GF_PLANE_TOGGLE_ON);
+}
+
+static void StartMenu_DestroyCursor(StartMenuTaskData *startMenu) {
+    sub_0200AEB0(startMenu->gfxResObj[GF_GFX_RES_TYPE_CHAR]);
+    sub_0200B0A8(startMenu->gfxResObj[GF_GFX_RES_TYPE_PLTT]);
+    for (u16 i = 0; i < 4; ++i) {
+        Destroy2DGfxResObjMan(startMenu->gfxResMan[i]);
+    }
+    SpriteList_Delete(startMenu->spriteList);
+}
+
+static void Task_StartMenu_WaitFade(TaskManager *taskManager) {
+    if (IsPaletteFadeFinished()) {
+        FieldSystem *fieldSystem = TaskManager_GetFieldSystem(taskManager);
+        StartMenuTaskData *startMenu = (StartMenuTaskData *)TaskManager_GetEnvironment(taskManager);
+
+        sub_0203C38C(startMenu, fieldSystem);
+        fieldSystem->unkD2_0 = 0;
+        startMenu->atexit_TaskFunc(taskManager);
+        startMenu->state = START_MENU_STATE_WAIT_APP;
+    }
+}
+
+static void Task_StartMenu_WaitApp(TaskManager *taskManager) {
+    FieldSystem *fieldSystem = TaskManager_GetFieldSystem(taskManager);
+    StartMenuTaskData *startMenu = (StartMenuTaskData *)TaskManager_GetEnvironment(taskManager);
+
+    if (!FieldSystem_ApplicationIsRunning(fieldSystem)) {
+        startMenu->atexit_TaskFunc(taskManager);
+        if (startMenu->state == START_MENU_STATE_RETURN && sub_020183F0(&fieldSystem->unk_10C) == TRUE) {
+            startMenu->state = START_MENU_STATE_10;
+        }
+    }
+}
+
+void StartMenu_SetChildProcReturnTaskFunc(StartMenuTaskData *startMenu, TaskFunc func) {
```

Comment:
You can probably just make this StartMenu_SetExitTask or ReturnTask or smth

Reply:
i decline, leaving as-is

Reply:
it's genuinely a kinda ugly name tbh, needlessly long with redundant info

Reply:
my opinion is unchanged